RecoParticleFlow/PFClusterProducer/src/Cluster3DPCACalculator.cc

```diff
@@ -0,0 +1,132 @@
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Utilities/interface/isFinite.h"
+
+#include <cmath>
+#include <unordered_map>
+
+#include "vdt/vdtMath.h"
+
+#include "RecoParticleFlow/PFClusterProducer/interface/PFCPositionCalculatorBase.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecHitFwd.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecHit.h"
+
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
+
+#include "TPrincipal.h"
```

Comment:
Looking at the `TPrincipal` code I couldn't find any obvious use of globals. 

However, the code generated by calling `TPrincipal::MakeCode` or `TPrincipal::MakeMethods` is not safe since it uses `static`. As far as I can tell, those two methods were not used in this pull request.